osu!stream/Main.cs

```diff
@@ -48,7 +50,16 @@ public Application()
         }
 
         public override void OnBackPressed() {
-            return;
+            if (Director.IsTransitioning) return;
+            
+            if(Director.CurrentOsuMode == OsuMode.Play)
+                if (!((Player)Director.CurrentMode).IsPaused)
+                    (Director.CurrentMode as Player)?.Pause();
+                else if(!(bool)(Director.CurrentMode as Player)?.menu.Failed)
+                    (Director.CurrentMode as Player)?.Resume();
+
+            if (Director.CurrentOsuMode == OsuMode.MainMenu)

```

Comment:
Is this really a thing? I'm pretty sure it shouldn't be.

Reply:
I tested it and it seems to work as intended, pausing when you press, unpausing when you press again, and not letting you unpause when you fail, i can double check again and record a vid if nessesary

Reply:
nvm i guess it didnt work fully and i was blind, resume doesnt seem to work properly, ill just press the continue button in the menu automatically

Reply:
Resolved in 5256813

Reply:
https://youtu.be/hlj7Lm1VJho
video here

Reply:
The issue is that a mobile app should never be ending the process. Especially on iOS. On android there should be a specific call to handle this that isn't "end execution", I think.

Reply:
i pulled it from here, but it seems there might be a better way https://stackoverflow.com/questions/29257929/how-to-terminate-a-xamarin-application

Reply:
and also this code is android-only, so it shouldnt be in effect on iOS

Reply:
What is listed on that page looks more correct. It means that application won't be unloaded from memory, which is what you want.

Reply:
The closest things that dont cause issues with assets unloading are `System.Diagnostics.Process.GetCurrentProcess().CloseMainWindow();` which just closes the main window instead of stopping execution (which is p much the same thing cus we reinitialize gamebase on re-load anyway)

Reply:
I wish i knew what the home button did so i could just do the same with the back button here but i have no idea how

Reply:
If this is intended to fix asset (vRAM) reloading then you will need to force quit the process. But I thought this was resolved via #26?

Reply:
Pressing the home button then going back was resolved via #26, but it seems that pressing the home button and pressing the back button do different things, so i had the back button disabled, when i dont disable the back button it seems to unload things in a way that even after force reloading *all* assets, everything is a white box, if i can replicate the home button behavior under the back button, it should fix it, and i do eventually want to allow the game to pause and continue properly when you minimize the app, so having one common "minimize" would be nice

Reply:
After a LOT of fiddling with stuff i got live tabbing in and out to work (turns out the engine can support it, i just have to hand hold it a bit with texture loading/unloading, which was easy enough)